src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1012,6 +1012,11 @@ private bool TryCreateSegmentForOperation(ODataPathSegment previousSegment, stri
                 bindingType = (previousSegment is EachSegment) ? previousSegment.TargetEdmType : previousSegment.EdmType;
             }
 
+            if (identifier != null && identifier.Length >= 1 && identifier[0] == ':' && bindingType != null)

```

Comment:
>identifier != null && identifier.Length >= 1 [](start = 16, length = 44)

Is this the same as !String.IsNullOrEmpty(identifier)? Is one better than the other?

Reply:
Yes. !String.IsNullOrEmpty(identifier) is better of readability.